src/ast.rs

```diff
@@ -1437,3 +1297,200 @@ mod tests {
         assert_eq!(VariableBlockType::InOut.to_string(), "InOut");
     }
 }
+
+pub struct AstFactory {}
+
+impl AstFactory {
+    /// creates a new if-statement
+    pub fn new_if_statement(
```

Comment:
I think create_.. is better here, you are also using them already below.

Reply:
done